src/friend_list.rs

```diff
@@ -1,115 +1,105 @@
-            (num_friends * mem::size_of::<FriendKey>() as u32) << 14 | ((0 & 0xF) << 10) | 0x2;
+        *cmdbuf.wrapping_add(4) = ((num_friends * mem::size_of::<FriendKey>() as u32) << 14) | 0x2;
         *cmdbuf.wrapping_add(5) = &friend_keys[0] as *const _ as u32;
         *cmdbuf.wrapping_add(6) =
             (num_friends * mem::size_of::<FriendInfo>() as u32) << 4 | 0x8 | 0b100;
         *cmdbuf.wrapping_add(7) = &mut friend_info[0] as *mut _ as u32;
 
-        _ = ctru_sys::svcSendSyncRequest(handle);
+        panic_if_failed!(ctru_sys::svcSendSyncRequest(handle));
+        panic_if_failed!(*cmdbuf.wrapping_add(1));
 
-        if *cmdbuf.wrapping_add(1) != 0 {
-            panic!("Something went wrong")
+        for i in 0..num_friends {
+            let pid: u32 = friend_keys[i as usize].principalId;
+            let mii_bytes: [u8; 0x5C] =
+                // There seems to be a bug in how FriendInfo is deserialized.
+                // The Mii data is simply incorrect, the version should always be 3, but it isn't
+                // Here I'm recreating the _bindgen_opaque_blob approach we had in earlier versions
+                mem::transmute::<_, [u8; mem::size_of::<FriendInfo>()]>(friend_info[i as usize])
+                    [128..220]
+                    .try_into()
+                    .unwrap();
+
+            match MiiData::from_bytes(mii_bytes) {
+                Ok(mii) => _ = friend_map.insert(pid, mii),
+                Err(e) => println!("{:#?}", e),
+            }
         }
-
-        (friend_keys.clone(), friend_info, num_friends)
     }
 }
 
-/*
-Result FRD_GetFriendInfo(FriendInfo *infos, const FriendKey *friendKeyList, u32 count, bool maskNonAscii, bool profanityFlag)
-{
-    Result ret = 0;
-    u32 *cmdbuf = getThreadCommandBuffer();
-
-    cmdbuf[0] = IPC_MakeHeader(0x1A,3,4); // 0x1A00C4
-    cmdbuf[1] = count;
-    cmdbuf[2] = (u32)maskNonAscii;
-    cmdbuf[3] = (u32)profanityFlag;
-    cmdbuf[4] = IPC_Desc_StaticBuffer(count * sizeof(FriendKey), 0);
-    cmdbuf[5] = (u32)friendKeyList;
-    cmdbuf[6] = IPC_Desc_Buffer(count * sizeof(FriendInfo), IPC_BUFFER_W);
-    cmdbuf[7] = (u32)infos;
-
-    if (R_FAILED(ret = svcSendSyncRequest(frdHandle))) return ret;
-
-    return (Result)cmdbuf[1];
+unsafe fn get_my_info(friend_map: &mut MiiMap, handle: Handle) {
+    unsafe {
+        let cmdbuf = ctru_sys::getThreadCommandBuffer();
+        *cmdbuf = 0x00050000;
+        panic_if_failed!(ctru_sys::svcSendSyncRequest(handle));
+        panic_if_failed!(*cmdbuf.wrapping_add(1));
+        let pid: u32 = *cmdbuf.add(2);
+
+        let cmdbuf = ctru_sys::getThreadCommandBuffer();
+        *cmdbuf = 0x000A0000;
+        panic_if_failed!(ctru_sys::svcSendSyncRequest(handle));
+
+        let mut mii = [0; _];
+        let mut idx = 0usize;
+        for i in 2..25 {
+            for v in (*cmdbuf.wrapping_add(i)).to_le_bytes() {
+                mii[idx] = v;
+                idx += 1;
+            }
+        }
```

Comment:
```suggestion
        let mut mii = [0; _];
        copy_nonoverlapping(
            cmdbuf.wrapping_add(2) as *mut _,
            mii.as_mut_ptr(),
            mii.len(),
        );
```
if it works in regards to endianness and stuff, just doing a memcpy is probably better